x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java

```diff
@@ -489,6 +494,40 @@ public void expandGroupIds(List<String> groupIds,  ActionListener<Set<String>> l
                 , client::search);
     }
 
+    public void findJobsWithCustomRules(ActionListener<List<Job>> listener) {
+        String customRulesPath = Strings.collectionToDelimitedString(Arrays.asList(Job.ANALYSIS_CONFIG.getPreferredName(),
+                AnalysisConfig.DETECTORS.getPreferredName(), Detector.CUSTOM_RULES_FIELD.getPreferredName()), ".");
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder()
+                .query(QueryBuilders.nestedQuery(customRulesPath, QueryBuilders.existsQuery(customRulesPath), ScoreMode.None))
+                .size(10000);
```

Comment:
This will have to be extracted to a constant and be used by all searches. I think we should also think how to refactor the code that does the job search into something that is reused as a few methods are doing such searches.